tools/stats/stats_analyzer.cpp

```diff
@@ -15,19 +15,340 @@
 #include "stats_analyzer.h"
 
 #include <algorithm>
+#include <cassert>
+#include <cstring>
 #include <iostream>
 #include <sstream>
 #include <vector>
 
+#include "spirv/1.2/spirv.h"
 #include "source/enum_string_mapping.h"
 #include "source/opcode.h"
+#include "source/operand.h"
 #include "source/spirv_constant.h"
-#include "spirv/1.1/spirv.h"
 
 using libspirv::SpirvStats;
 
 namespace {
 
+// Returns all SPIR-V v1.2 opcodes.
+std::vector<uint32_t> GetAllOpcodes() {
+  return std::vector<uint32_t>({
+    SpvOpNop,
```

Comment:
Would be slightly nicer to generate this automatically.  But this is good enough for your stats.